library/network_connections.py

```diff
@@ -837,6 +859,40 @@ def connection_create(self, connections, idx, connection_current=None):
                     NM.SETTING_WIRED_SPEED, connection["ethernet"]["speed"]
                 )
 
+            # Handle ethtool features
+            # If NM supports ethtool settings:
+            # Set them accordingly
+            # If NM does not support ethtool settings:
+            # (a) if some are specified, fail
+            # (b) if none are specified, ignore them
+            ethtool_features = connection["ethtool"]["features"]
+            try:
+                s_ethtool = NM.SettingEthtool.new()
+            except AttributeError:
+                # NM is too old to support SettingEthtool
+                s_ethtool = False
+
+            if s_ethtool:
+                con.add_setting(s_ethtool)
+                for feature, setting in ethtool_features.items():
+                    nm_feature = nm_provider.get_nm_ethtool_feature(feature)
+
+                    if setting is None:
+                        # Only set features to default value that NM knows
+                        if nm_feature:
+                            s_ethtool.set_feature(nm_feature, NM.Ternary.DEFAULT)
+                    else:
+                        if not nm_feature:
+                            raise MyError(
+                                "ethtool feature %s not support by NM" % feature
+                            )
+                        if setting:
+                            s_ethtool.set_feature(nm_feature, NM.Ternary.TRUE)
+                        else:
+                            s_ethtool.set_feature(nm_feature, NM.Ternary.FALSE)
+            elif set(ethtool_features.values()) != set([None]):
```

Comment:
is this pythonic? Seems hard to read. Why not
```
    if not all([f is None for f in ethtool_features.values()]):
```
(untested)

Reply:
awesome advice, I think I would like it even more as:

```python
if any([value is not None for value in ethtool_features.values()])
```

Will change it.

Reply:
and "any()" is faster than "not all()", because it can stop iterating the list comprehension once the result is decided...